server/src/microservices/processors/metadata-extraction.processor.ts

```diff
@@ -178,165 +110,110 @@ export class MetadataExtractionProcessor {
+    const { latitude, longitude } = exifData;
+    if (!this.reverseGeocodingEnabled || !longitude || !latitude) {
+      return exifData;
     }
+
+    try {
+      const geocode = await this.geocodingRepository.reverseGeocode({ latitude, longitude });
+      return { ...exifData, ...geocode };
+    } catch (error: any) {
+      this.logger.warn(
+        `Unable to run reverse geocoding due to ${error} for asset ${asset.id} at ${asset.originalPath}`,
+        error?.stack,
+      );
+    }
+    return exifData;
   }
 
-  private extractDuration(duration: number | string | null) {
-    const videoDurationInSecond = Number(duration);
-    if (!videoDurationInSecond) {
+  private async exifData(asset: AssetEntity): Promise<[ExifEntity, ImmichTags]> {
+    const mediaTags = await exiftool.read<ImmichTags>(asset.originalPath).catch((error: any) => {
+      this.logger.warn(`error reading exif data (${asset.id} at ${asset.originalPath}): ${error}`, error?.stack);
       return null;
-    }
+    });
 
-    return Duration.fromObject({ seconds: videoDurationInSecond }).toFormat('hh:mm:ss.SSS');
+    const sidecarTags = asset.sidecarPath
+      ? await exiftool.read<ImmichTags>(asset.sidecarPath).catch((error: any) => {
+          this.logger.warn(`error reading exif data (${asset.id} at ${asset.sidecarPath}): ${error}`, error?.stack);
+          return null;
+        })
+      : null;
+
+    const stats = fs.statSync(asset.originalPath);
+
+    const tags = { ...mediaTags, ...sidecarTags };
+
+    const exifDate = (dt: ExifDateTime | string | undefined) => (dt instanceof ExifDateTime ? dt?.toDate() : null);
+    const zone = (dt: ExifDateTime | string | undefined) => (dt instanceof ExifDateTime ? dt.zone : null);
+
+    const parseFocalLength = (focalLength: string | undefined) =>
+      focalLength ? parseInt(focalLength.split(' ')[0]) : null;
+
+    const validate = <T>(value: T): T | null => (typeof value === 'string' ? null : value ?? null);
+
+    return [
+      <ExifEntity>{
+        // altitude: tags.GPSAltitude ?? null,
+        assetId: asset.id,
+        dateTimeOriginal: exifDate(tags.CreateDate) ?? asset.fileCreatedAt,
+        exifImageHeight: validate(tags.ImageHeight),
+        exifImageWidth: validate(tags.ImageWidth),
+        exposureTime: tags.ExposureTime ?? null,
+        fileSizeInByte: stats.size,
+        fNumber: validate(tags.FNumber),
+        focalLength: parseFocalLength(tags.FocalLength),
+        fps: validate(tags.VideoFrameRate),
+        iso: validate(tags.ISO),
+        latitude: validate(tags.GPSLatitude),
+        lensModel: tags.LensModel ?? null,
+        longitude: validate(tags.GPSLongitude),
+        make: tags.Make ?? null,
+        model: tags.Model ?? null,
+        modifyDate: exifDate(tags.ModifyDate) ?? asset.fileModifiedAt,
+        orientation: validate(tags.Orientation)?.toString() ?? null,
+        timeZone: zone(tags.OriginalCreateDateTime),
```

Comment:
no need for this, just use https://github.com/photostructure/exiftool-vendored.js/blob/6408b855b80fe322a8dda3f1c4807f1cc1333844/src/Tags.ts#L4570